design/20240122.scarf.md

```diff
@@ -0,0 +1,62 @@
+
+## Motivation
+
+### Goals
+
+- Discontinue using the name "Jetstack" in all container image downloads.
+- Continue to provide non-breaking changes for those already using the existing "Jetstack" container image download locations.
+- Gain the freedom to change image repositories with ease when necessary.
+- Improve observability and reporting to maintainers of how cert-manager is downloaded to serve its users better.
+
+## Proposal
+
+- Obtain a new custom "download" domain through the CNCF to be used for fronting all binary downloads.
+- The creation of a free (OSS tier) Scarf account will be configured and managed by the cert-manager maintainers.
+- Update documentation referencing "jetstack" binary paths e.g. quay.io/jetstack/cert-manager-controller, and replace with the new download domain.
+- Update helm charts referencing "jetstack" binary paths, replacing with the new download domain.
+- Update code referencing "jetstack" binary paths, replacing with the new download domain.
+- Add Scarf pixels to selective documentation pages, giving us insight into which pages are most useful or areas to focus on for improvement.
+- Automate regular analytics gathering leveraging the Scarf API to publish relevant stats and info publically. E.g.
+  - Region
+  - Operating System
+  - Container Tags / Versions
+  - Container Runtimes
+
+
+### How will this change impact our users?
+
+Images and binaries should continue to be accessed from their existing locations. Therefore, there should be no impact on any existing downloads, automation, or mirroring.
+Going forward, we would encourage users to use the new download paths by specifying the new domain in the documentation.
+
+Any users downloading from secure environments with limited internet connections through firewall restrictions will need to add "allowed" rules for the Scarf gateway domain in addition to any existing rules for the image repository, such as quay.io. These should be clearly documented.
+
+### Known issues/limitations
+- Currently, the Scarf service only allows for custom domains and doesn't include custom paths. When speaking with members of the Scarf organisation, this is due to a technical limitation as the path is used in the image identification/verification process. Scarf is investigating a workaround; however, we may need to consider an additional hosting location/service to allow us to remove "jetstack" from the download path. An additional hosting location will increase existing maintenance and deployment process overheads.
```

Comment:
### Risks

Some risks which we should consider.

* Outages of the gateway service: 
   -  https://github.com/keptn/lifecycle-toolkit/issues/1431#issuecomment-1556575315
   
* Ad Blocking software blocking the scarf proxy (and vanity domains that use the scarf CNAME)
   - https://github.com/GhostWriters/DockSTARTer/issues/1715
   - https://github.com/easylist/easylist/issues/18395
   - https://discourse.pi-hole.net/t/apply-pi-hole-blocking-to-cnames/25445